jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -362,6 +362,44 @@ private static PrintStream createConsole(FileDescriptor desc) {
 	/*[ENDIF] Sidecar18-SE-OpenJ9 */
 	/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
 
+	/*[IF JFR_SUPPORT]*/
+	static void initJFR() {
+		boolean enableJFRDebug = false;
+		String enableJFRDebugProp = internalGetProperties().getProperty("enable.j9internal.jfr.debug");
+		if (null != enableJFRDebugProp) {
+			if (enableJFRDebugProp.equalsIgnoreCase("no")) {
+				enableJFRDebug = false;
+			} else if (enableJFRDebugProp.equalsIgnoreCase("yes")) {
+				enableJFRDebug = true;
+			}
+		}
+		String jfrCMDLineOption = com.ibm.oti.vm.VM.getjfrCMDLineOption();
+		if (null != jfrCMDLineOption) {
+			try {
+				Class<?> dcmdStartClass = Class.forName("jdk.jfr.internal.dcmd.DCmdStart");
+				Constructor<?> constructor = dcmdStartClass.getDeclaredConstructor();
+				constructor.setAccessible(true);
+				Object dcmdStartInstance = constructor.newInstance();
+				/*[IF JAVA_SPEC_VERSION == 17]*/
```

Comment:
@TobiAjila - Actually JDK11 and JDK17 execute method has different parameter list. In 17, the parsing is done inside execute while 11 expects the parsed options.

J17 - `public final String[] execute(String source, String arg, char delimiter)`

J11 - `public String execute(String name, String[] settings, Long delay, Long duration, Boolean disk, String path, Long maxAge, Long maxSize, Boolean dumpOnExit, Boolean pathToGcRoots)`

Should I wait for this to be merged or shall I add code for 11 in this same PR?

Reply:
You can add the JDK11 code here